docs/proposals/PROPOSAL_B.md

```diff
@@ -0,0 +1,534 @@
+1. [ ] As a system runtime administrator, I want to use annotations to schedule for the appropriate resources.
+1. [x] As a system runtime administrator, the runtime should not need to know about all possible types of hardware.
+   Perhaps hooks could be added for users to inject their own image selection criteria on a given host, or annotations could be injected.
+
+### Deployment Engineer
+
+1. [ ] As a deployment engineer, I want to parse an image index and find the “optimal” image for the cluster node I am aiming to run the image on.
+That includes being able to:
+   - Discover the image that fits the selected host.
+   - Find the best match from the nodes and images I have available.
+   - Determine that the image is not fitting the selected host.
+1. [ ] As a deployment engineer, I want the image compatibility check to be performed without downloading or executing the referenced image layers.
+1. [x] As a deployment engineer, I should be able to add the compatibility to images already being used in production, especially for the images released before image compatibility wg was created.
+1. [x] As a deployment engineer, I want to be able to specify the version and variant of an application or other user specific configuration (e.g. MPI), and not only hardware/kernel details in the compatibility specification.
+1. [ ] As a deployment engineer, I want my compatibility spec runtime to be able to select the best possible runtime available on a node (e.g. runc vs nvidia vs wasm).
+1. [ ] As a deployment engineer, I want to be able to rank the images in a multi-platform image so that the runtime can know which one to choose when more than one image is compatible with the runtime environment.
+1. [x] As a deployment engineer, I want to reuse community projects so that I don't duplicate and integrate the functionality of the already existing tools.
+
+### Registry Maintainer
+
+1. [x] As a registry user or operator I want to have a common way of inspecting information about image compatibility to enable users to find an image that best matches their system. (_That can be achieved by the tool provided by the OCI group_)
+1. [x] As a registry operator (or user that has no control over their registry implementation), I want any compatibility changes to not depend on registry server changes or upgrades.
+
+### OCI Specification Maintainer
+
+1. [x] As a spec maintainer, I want the solution to avoid breaking other specs (confidential images, image signing, existing implementations for runtimes picking images).
+1. [x] As a spec maintainer, I don't want the spec to update for new hardware devices, kernel releases, or other external dependencies.
+1. [x] As a spec maintainer, I don't want to overlap significantly with solutions from other specs (like SBOMs).
+
+### Security Administrator
+
+1. [x] As a security administrator, I want predictable behavior from runtimes, which does not change based on unsigned content.
```

Comment:
This is because the spec doesn't do runtime selection, correct? Otherwise, we'd need to document how to handle an image that is signed but an artifact that is either unsigned for signed by a different entity.

Reply:
> This is because the spec doesn't do runtime selection, correct? 

Correct.